src/lib/stream.rs

```diff
@@ -278,6 +320,25 @@ impl<'a> CodedInputStream<'a> {
         self.read_raw_varint64().map(|v| v as u32)
     }
 
+    #[cfg(target_endian = "little")]
+    pub fn read_raw_little_endian32(&mut self) -> ProtobufResult<u32> {
+        let mut bytes = [0u8; 4];
+        try!(self.read_raw_bytes_immediate(&mut bytes));
+        Ok(
+            unsafe { mem::transmute(bytes) }
+        )
+    }
+
+    #[cfg(target_endian = "little")]
+    pub fn read_raw_little_endian64(&mut self) -> ProtobufResult<u64> {
+        let mut bytes = [0u8; 8];
+        try!(self.read_raw_bytes_immediate(&mut bytes));
+        Ok(
+            unsafe { mem::transmute(bytes) }
```

Comment:
Nah, rustc generates:

```llvm
  %a = alloca [8 x i8]
```

... without an alignment, so it's up to the backend to choose a default.  Which, may or may not be word-aligned, so this is probably not safe on all platforms (though I think it probably happens to be on x64).  I'll change it.